src/bootstrap/src/lib.rs

```diff
@@ -69,16 +69,59 @@ const LLVM_TOOLS: &[&str] = &[
 /// LLD file names for all flavors.
 const LLD_FILE_NAMES: &[&str] = &["ld.lld", "ld64.lld", "lld-link", "wasm-ld"];
 
+#[derive(Clone, Debug)]
+pub struct ConfigChange {
+    /// Represents the ID of PR caused major change on bootstrap.
+    pub change_id: usize,
+    pub severity: Severity,
+    /// Provides a short summary of the change that will guide developers
+    /// on "how to handle/behave" in response to the changes.
+    pub summary: &'static str,
+}
+
+#[derive(Clone, Debug)]
+pub enum Severity {
+    Info,
```

Comment:
It would be useful to document here what do these levels mean, and perhaps provide some examples (like "info = giving notice about new options, no need to modify your local config", "warning = you might need to modify your local config").

Reply:
Does https://github.com/rust-lang/rust/pull/117815/commits/a4f05c6e0e54f879de17b0c2b8deecac17f87fe5 help?

Reply:
> and perhaps provide some examples

The existing change list is already a good example I think.

Reply:
Yes, great! Thank you.